client/my-sites/themes/helpers.ts

```diff
@@ -23,7 +39,20 @@ function appendActionTracking( option, name ) {
 	} );
 }
 
-export function getAnalyticsData( path, { filter, vertical, tier, site_id } ) {
+export function getAnalyticsData(
+	path: string,
+	{
+		filter,
+		vertical,
+		tier,
+		site_id,
+	}: {
+		filter: string | undefined;
+		vertical: string | undefined;
+		tier: string | undefined;
+		site_id: string | undefined;
+	}
```

Comment:
I don't think declaring types inline is a good idea. What is that second argument all about? Options? I'd rather be explicit here. It makes the code easier to reason about. We can keep the parameter destructuring, but at least adding the type annotation might make things easier to understand.

Suggestion:

```ts
interface GetAnalyticsDataOptions {
  filter?: string
  vertical?: string
  tier?: string
  site_id?: string
}
```

Reply:
> I don't think declaring types inline is a good idea. What is that second argument all about? Options? I'd rather be explicit here.

I didn't realize there was a meaningful difference between 

```
interface GetAnalyticsDataOptions {
  filter?: string
}
```

and

```
interface GetAnalyticsDataOptions {
  filter: string | undefined
}
```

Would you mind elaborating more on what each implementation conveys?